src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp

```diff
@@ -17,337 +17,477 @@
+                        else
+                            announcer->CastSpell(me->GetVictim(), SPELL_RAISE_ARELAS);
+                    }
+                }
+                else
+                {
+                    // Phase 2 only
+                    events.ScheduleEvent(EVENT_DESECRATION, 5s);
+                    events.ScheduleEvent(EVENT_GHOUL_EXPLODE, 8s);
+
+                    // Army of the Dead
+                    // disabling movement temporarily so the spell wont get interrupted
+                    me->SetControlled(true, UNIT_STATE_ROOT);
+                    DoCast(SPELL_ARMY_DEAD);
                 }
             }
+        }
 
-            if (!me->HasUnitState(UNIT_STATE_ROOT) && !me->HealthBelowPct(1))
-                DoMeleeAttackIfReady();
+        void SetData(uint32 type, uint32 /*data*/) override
+        {
+            if (type == 1)
+                achievementCredit = false;
         }
 
-        void DamageTaken(Unit* /*pDoneBy*/, uint32& uiDamage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+        void SpellHit(WorldObject* /*target*/, SpellInfo const* spellInfo) override
         {
-            if (uiDamage > me->GetHealth() && uiPhase <= PHASE_SKELETON)
+            if (spellInfo->Id == SPELL_BLACK_KNIGHT_RES)
             {
-                uiDamage = 0;
-                me->SetHealth(0);
-                me->AddUnitState(UNIT_STATE_ROOT | UNIT_STATE_STUNNED);
-                summons.DespawnAll();
-                switch (uiPhase)
+                // TODO: According to sniffs, The Black Knight should update
+                // creature template to another entry, not just change his display id
+                // But the other templates are missing correct damage modifiers
+                me->SetFullHealth();
+                me->SetStandState(UNIT_STAND_STATE_STAND);
+                if (uiPhase == PHASE_UNDEAD)
                 {
-                    case PHASE_UNDEAD:
-                        me->SetDisplayId(MODEL_SKELETON);
-                        break;
-                    case PHASE_SKELETON:
-                        me->SetDisplayId(MODEL_GHOST);
-                        break;
+                    me->SetDisplayId(MODEL_SKELETON);
+                    Talk(SAY_PHASE_2);
+                }
+                else if (uiPhase == PHASE_SKELETON)
+                {
+                    me->SetDisplayId(MODEL_GHOST);
+                    SetEquipmentSlots(false, EQUIP_UNEQUIP, EQUIP_NO_CHANGE, EQUIP_NO_CHANGE);
+                    Talk(SAY_PHASE_3);
                 }
-                bEventInProgress = true;
+                ++uiPhase;
+                LoadEvents();
             }
         }
 
-        void JustDied(Unit* /*killer*/) override
+        void MovementInform(uint32 type, uint32 id) override
         {
-            DoCast(me, SPELL_KILL_CREDIT);
+            if (type != POINT_MOTION_TYPE)
```

Comment:
POINT_MOTION_TYPE is a undeclared value per the nopch check. Maybe a missing include that may be causing it.